### ISSUE TITLE
Mumble overlay does not work

### ISSUE THREAD

**Issue by 3JlOy-PYCCKUi** [mumble-voip/mumble#5932]
### Description

There is an error, related to `LD_PRELOAD`, when trying to run the overlay.
My distro is Arch Linux, mumble installed from official repositories.

### Steps to reproduce

1. Try to start mumble overlay

### Mumble version

1.4.287

### Mumble component

Client

### OS

Linux

### Reproducible?

Yes

### Relevant log output

```shell
$ LANG=C.UTF-8 mumble-overlay glxgears
set
set
Unable to find symbol malloc. Aborting.
/usr/bin/mumble-overlay: line 69:  5583 Aborted                 LD_PRELOAD="${MUMBLE_PRELOAD} ${LD_PRELOAD}" "${@}"
```

**Comment by Krzmbrzl** [comment]
@davidebeatrici any idea?

**Comment by Krzmbrzl** [comment]
@3JlOy-PYCCKUi could you try running the overlay + application through gdb, so we can get a backtrace of the crash?

**Comment by Krzmbrzl** [comment]
Ah yes, right. `mumble-overlay` is only a wrapper script... :facepalm:

**Comment by radioactiveman** [comment]
Have you installed `lib32-glibc`? It's an opt-dependency for Mumble and is needed for `mumble-overlay`.

https://archlinux.org/packages/community/x86_64/mumble/

**Comment by Krzmbrzl** [comment]
@3JlOy-PYCCKUi some things that come to my mind:
The overlay not finding `glPopClientAttrib` most likely means that the target program was not linked against OpenGL, which is a requirement for the overlay to work. 

What goes on in the minetest case, I don't know.  But for your original problem it might bw that the program uses a custom memory allocator instead of `malloc`, which would explain why that symbol can't be found...

**Comment by Krzmbrzl** [comment]
Oops, I accidentally deleted the comment of OP confirming that lib32-glibc is indeed installed.

**Comment by Krzmbrzl** [comment]
Afaik steam games use Vulkan, which is not supported by our overlay.

For those games that still don't work, are the error messages as before?

**Comment by Krzmbrzl** [comment]
gzdoom and xeyes probably don't use OpenGL and therefore it is expected that they don't work

**Comment by Green-Sky** [comment]
vulkan layer when?

**Comment by Krzmbrzl** [comment]
When someone finds time and motivation to implement it :P
There is a dedicated ticket somewhere here on GitHub - it's rather old already though (iirc) :thinking:

**Comment by cedarmax** [comment]
Is there any way to just make the Mumble overlay always present, regardless of a game being open?  I have the same issue.

**Comment by Krzmbrzl** [comment]
I don't think so. The overlay hooks into the graphics call of the game to do its rendering, so you have to have a game running for it to work.

**Comment by Green-Sky** [comment]
> I don't think so. The overlay hooks into the graphics call of the game to do its rendering, so you have to have a game running for it to work.

except for vulkan, which could, if implemented, be installed as a systemwide layer.

**Comment by cedarmax** [comment]
> I don't think so. The overlay hooks into the graphics call of the game to do its rendering, so you have to have a game running for it to work.

It is suggested here that this is possible: https://github.com/mumble-voip/mumble/issues/1273#issuecomment-46081763
On Windows it is possible to have the overlay always present.  I'm trying to get the overlay to work with TF2 but the issue according to others seems to be that mumble is 64 bits, steam is 64 bits, but TF2 is 32 bits.  Not that the overlay works in any other game, anyways.

**Comment by r3k2** [comment]
Any news on this? I just installed mumble and looking for the overlay to work on vulkan as in like everything is vulkan now a days :sweat_smile:

**Comment by cedarmax** [comment]
> Any news on this? I just installed mumble and looking for the overlay to work on vulkan as in like everything is vulkan now a days 😅
I updated to Ubuntu 24.04 LTS with Mumble 1.5.517.  The overlay is still broken.

**Comment by cedarmax** [comment]
Doesn't matter what I try to run mumble-overlay with, I get the same output:

cedar@desktop:~$ mumble-overlay xeyes
set
xeyes: symbol lookup error: /usr/lib/x86_64-linux-gnu/mumble/libmumbleoverlay.so: undefined symbol: glPopClientAttrib

The official website claims support for 32-bit and 64-bit OpenGL and Vulkan applications on Linux:

https://www.mumble.info/documentation/user/in-game-overlay/

Please show me how!

**Comment by Krzmbrzl** [comment]
The overlay doesn't support Vulkan (#2134) nor DX12.
Beyond that, you might be able to find some helpful information in the following issues: https://github.com/mumble-voip/mumble/issues?q=is%3Aissue%20state%3Aclosed%20glPopClientAttrib

**Comment by radioactiveman** [comment]
> The official website claims support for 32-bit and 64-bit OpenGL and Vulkan applications on Linux:
https://www.mumble.info/documentation/user/in-game-overlay/

> The overlay doesn't support Vulkan (https://github.com/mumble-voip/mumble/issues/2134) nor DX12.

Why do the support tables in the docs mention Vulkan then?
I guess "Direct3D 12 - does not display" should also be red in both tables.

**Comment by Krzmbrzl** [comment]
I have no idea :shrug:

**Comment by davidebeatrici** [comment]
The table in general is wrong:

<img width="1219" height="538" alt="Image" src="https://github.com/user-attachments/assets/7856908d-8a3f-4d1d-b060-81e629471d1c" />

- Overlay support for DirectX and OpenGL has been added way before 1.3.0.
- Inconsistencies between 64 bit and 32 bit.
